### ISSUE TITLE
Where's DeepLabv3+ model Decoder?

### ISSUE THREAD

**Issue by Albert-learner** [charlesCXK/SA-Gate#26]
Hello, I'm trying to study about your code. But when I analysis your code, I can't find DeepLabV3+ Decoder in your code. Could you please tell me where the DeepLabV3+ model's Decoder?

**Comment by Albert-learner** [comment]
And I have another question. You're using one of Semantic Segmentation model, DeepLab V3+. But I know that the result of Semantic Segmentation result is [height, width, channels], and apply argmax so that I could get one channel array. When I analyze the result of Semantic Segmentation ath NYU Depth V2 dataset, the result array(inference DeepLab V3+) consists of different min cost and max cost at each channel array. 
I'd like to know that why result array of each category's range is different.

**Comment by charlesCXK** [comment]
> Hello, I'm trying to study about your code. But when I analysis your code, I can't find DeepLabV3+ Decoder in your code. Could you please tell me where the DeepLabV3+ model's Decoder?

Hi, DeepLabV3+ decoder is defined [here](https://github.com/charlesCXK/RGBD_Semantic_Segmentation_PyTorch/blob/32b3f86822d278103a13ea6f93f9668d3b631398/model/SA-Gate.nyu/network.py#L149).

**Comment by charlesCXK** [comment]
> And I have another question. You're using one of Semantic Segmentation model, DeepLab V3+. But I know that the result of Semantic Segmentation result is [height, width, channels], and apply argmax so that I could get one channel array. When I analyze the result of Semantic Segmentation ath NYU Depth V2 dataset, the result array(inference DeepLab V3+) consists of different min cost and max cost at each channel array. I'd like to know that why result array of each category's range is different.

The absolute values of the result array do not make much sense. 
- During training, the result array is supervised by the CrossEntropy loss, which pushes the confidence of the correct category to be higher than other categories.  In the meanwhile, the SoftMax operation in CrossEntropy loss hopes that the confidence of the correct category is significantly different from the other categories. For other categories, we don't care what their confidence numbers are, just make them as low as possible.
- During inference, we simply select the category with the maximum confidence as the prediction, which is consistent with the training.
